lib/galaxy/webapps/galaxy/api/workflows.py

```diff
@@ -1005,34 +1003,27 @@ def _generate_invocation_bco(self, trans, invocation_id, **kwd):
 
         # pull in the user info from those who the history and workflow has been shared with
         contributing_users = [stored_workflow.user]
-        for association in trans.sa_session.query(model.StoredWorkflowUserShareAssociation):
-            if association.user not in contributing_users:
-                contributing_users.append(association.user)
-        for h_association in trans.sa_session.query(model.HistoryUserShareAssociation):
-            if h_association.user not in contributing_users:
-                contributing_users.append(h_association.user)
 
         # may want to extend this to have more reviewers.
         reviewing_users = [stored_workflow.user]
         encoded_workflow_id = trans.security.encode_id(stored_workflow.id)
         encoded_history_id = trans.security.encode_id(history.id)
         dict_workflow = json.loads(self.workflow_dict(trans, encoded_workflow_id))
 
-        # h_contents = self.history_contents_manager.contained(history)
-
         spec_version = kwd.get('spec_version', 'https://w3id.org/ieee/ieee-2791-schema/2791object.json')
 
         # listing the versions of the workflow for 'version' and 'derived_from'
         versions = []
-        workflow_versions = self.workflow_manager.get_stored_accessible_workflow(trans, encoded_workflow_id)
-        for i, w in enumerate(reversed(workflow_versions.workflows)):
+        for i, w in enumerate(reversed(stored_workflow.workflows)):
             version = {
                 'version': i,
                 'update_time': str(w.update_time),
-                'steps': len(w.steps),
+                'steps': w.step_count,
                 'version_id': trans.security.encode_id(w.stored_workflow.id),
             }
             versions.append(version)
+            if workflow == w:
+                break
         current_version = versions[-1]['version']
```

Comment:
I think this can be greatly simplified to:
```python
        current_version = None
        for i, w in enumerate(reversed(stored_workflow.workflows)):
            if workflow == w:
                current_version = i
```